packages/components/src/components/TaskTree/TaskTree.jsx

```diff
@@ -58,10 +66,17 @@ const TaskTree = ({
         const { reason, status } = getStatus(taskRunToUse);
         const { steps } = taskRunToUse.status || {};
         const expanded =
-          (!selectedTaskId && erroredTask?.metadata.uid === uid) ||
-          selectedTaskId === pipelineTaskName ||
-          (!erroredTask && !selectedTaskId && index === 0);
-        const selectDefaultStep = !selectedTaskId;
+          !hasExpandedTask &&
```

Comment:
Found this value tough to deconstruct. Perhaps worth adding a comment or two, or splitting up into more variables with explanatory names?

Reply:
Yeah there's a lot going on there, added comments for each branch. Let me know if that helps

Reply:
Looks good, thanks